src/snowflake/connector/connection.py

```diff
@@ -207,6 +207,7 @@ def __init__(self, **kwargs):
         self.messages = []
         self._async_sfqids = set()
         self._done_async_sfqids = set()
+        self.telemetry_enabled = False
```

Comment:
So customers will have to turn it on manually right?

Reply:
Yes. If they have telemetry enabled then this will send the exception information to Snowflake. If it is disabled then it won't get sent. However, we don't have flags for sending specific types of telemetry. Its either all telemetry info is sent or no telemetry info is sent

Reply:
Seems reasonable for now. If we start adding too much different types of telemetry it might be worth while to explore some form of layered approach in the design.